### DISCUSSION TITLE
Welcome to cookiecutter-openedx-devops Discussions!

### DISCUSSION THREAD

**Discussion by lpm0073** [lpm0073/cookiecutter-openedx-devops#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Soporte-Crezee** [comment]
Hi Mr. Lawrence !

I'm follow the tutorial, i already finish the cookiecutter cuestionarie, but when I get to this part:

git remote add origin https://github.com/lpm0073/openedx_devops.git

git push -u origin main

I can't do the push, I wrote my git user and my token but the following appears:

remote: Permission to lpm0073/openedx_devops.git denied to Soporte-Crezee.
fatal: unable to access 'https://github.com/lpm0073/openedx_devops.git/': The requested URL returned error: 403

**Reply by lpm0073** [reply]
the documentation is unclear. i'll work on that. the commands that you should execute are:

`git remote add origin https://github.com/[YOUR GITHUB ACCOUNT NAME]/openedx_devops.git`
`git push -u origin main`

**Reply by Soporte-Crezee** [reply]
Hello again !

I am at step "III. Build Your Open edX Backend" and when I run "terragrunt apply" on the path "./terraform/environments/{{ cookiecutter.environment_name }}/vpc", the command starts correctly but after Terraform has been initialized successfully, I get the following error:

Error: no matching Route53Zone found
│
│ with data.aws_route53_zone.environment_domain,
│ on route53.tf line 16, in data "aws_route53_zone" "environment_domain":
│ 16: data "aws_route53_zone" "environment_domain" {
│
╵
ERRO[0018] 1 error occurred:
        * exit status 1

![imagen_2022-04-19_175600154](https://user-images.githubusercontent.com/102400906/164114522-09571ce1-3805-4c11-bf72-f2fb903d6573.png) 
![imagen_2022-04-19_175630911](https://user-images.githubusercontent.com/102400906/164114555-3ac11824-d94b-4d31-990a-ff966801374c.png)

**Reply by lpm0073** [reply]
you should manually add an AWS Route53 hosted zone for your project's environment domain. If for example, you want host the LMS at the domain courses.myschool.edu, then note that the `root domain` is myschool.edu whereas the `environment domain` is courses.myschool.edu is the environment domain. The Terraform module expects the environment domain to already exist in your AWS Route53 hosted zones.

**Reply by Soporte-Crezee** [reply]
Hello again! and excuse me for my constant posts.

I am in step III "Build Your Open edX Backend ", when I execute the command "terragrunt run-all apply" in the corresponding folder, I get 3 errors, 2 are messages that indicate the following: "Error: Error acquiring the state lock" and the last message indicates: "Error: no matching EC2 Key Pair found".

I don't understand how to solve these problems, for the "state lock" messages I have tried the commands "terraform force-unlock <ID>" and "terraform force-unlock -force <ID>" but neither works. Regarding the third message, I honestly have no idea how to fix it.

Attached the images of these errors and I thank you for the great support you have given me.

![image](https://user-images.githubusercontent.com/102400906/165792074-03f62d80-b9ec-4993-936b-efc10fd5dbb5.png)

![image](https://user-images.githubusercontent.com/102400906/165792183-a8f4aa25-f8d5-406a-b2cf-2f8c4baaeabe.png)

![image](https://user-images.githubusercontent.com/102400906/165792285-63c8883f-4ced-4d7d-ae74-d6ddb52ab9c2.png)

![image](https://user-images.githubusercontent.com/102400906/165792394-eef5c7ad-3d11-47a7-ac6a-0fea0f56b0d6.png)

![image](https://user-images.githubusercontent.com/102400906/165792474-0bd6dbfd-0ca6-4aca-b0bd-d24c9ac2e68b.png)

![image](https://user-images.githubusercontent.com/102400906/165792579-14d72511-5074-4035-852e-5c78719cce40.png)

![image](https://user-images.githubusercontent.com/102400906/165792688-029379db-85d8-499d-a725-276e94302044.png)

![image](https://user-images.githubusercontent.com/102400906/165792798-6c06eb5b-b33f-47f7-859e-0b6e3e753671.png)

Additionally, i have these messages: 

![image](https://user-images.githubusercontent.com/102400906/165793125-3af1f714-a8ff-4165-bbac-639acc9b383d.png)

![image](https://user-images.githubusercontent.com/102400906/165793205-7926b3f3-6a7f-4085-8cdf-58e0e0b2a843.png)

**Reply by lpm0073** [reply]
this is great feedback, thank you!! :D  Please note the following:

1. "Error: Error acquiring the state lock". This is a Terraform error indicating that it either cannot access the AWS S3 bucket in which the state of your AWS account is persisted, or that the content of the DynamoDB table that is automatically created by Terraform is corrupted. The first step you should attempt is to delete the DynamoDB table (there will be one table per environment). Terraform will automatically replace the table and its contents during its "init" process. If that does not fix the problem then you should review the corresponding AWS S3 bucket, the name of which begins with a 12-digit number, like this, "320713933456-terraform-tfstate". Terraform creates this bucket for you automatically. If you see the bucket then just empty its contents. Terraform will automatically rebuild the database.
2. "Error: no matching EC2 Key Pair found". You see this error because you chose the Cookiecutter option to create an EC2 bastion, but you have not yet created an SSH key pair for the instance. You have to manually create the key pair from the EC2 console, and then afterwards you provide the name of the key as a Cookiecutter parameter.

**Reply by Soporte-Crezee** [reply]
Hi Mr. Lawrence! good Monday

I'm still on step 3, I tried to run the "terragrunt apply" command on the eks_clb_ec2 folder but I got the following error:
╷
│ Error: Unsupported argument
│
│ on main.tf line 26, in provider "kubernetes":
│ 26: load_config_file = false
│
│ An argument named "load_config_file" is not expected here.
╵
ERRO[0016] 1 error occurred:
        * exit status 1

Do you have any idea what could be happening? 
I add an image

![image](https://user-images.githubusercontent.com/102400906/166297950-a13df6aa-1458-41b4-96c9-12e2c9540988.png)

**Comment by lpm0073** [comment]
Confirming that I was able to duplicate your error above. Summarizing the steps to resolve:

- completely delete the repository that you created with the Cookiecutter
- re-create your repository from Cookiecutter using latest
- re-run the Terragrunt scripts

more detail: you've somehow gotten an older set of Terraform code (from around 2 months ago) running against a new installation of AWS EKS Elastic Kubernetes Service. The specific error you posted above is a line from the old source code, referencing a boolean flag, "load_config_file" that has since been deprecated.

**Reply by lpm0073** [reply]
it's good that you're reporting these hurdles. i'm sure others will have similar if not identical experiences. this forum is searchable on Google, which should help the community.

**Reply by Soporte-Crezee** [reply]
Hello Mr. Larence!

Thank you very much for your great help.

I did what you recommended, I deleted the repository with the old code and created a new one with the current code.

Everything goes very well, until I get to step 3 "III. Build Your Open edX Backend"

I am now applying the terragrunt commands on each element following the specified order, now I am in the "kubernetes_ingress_clb" folder and run terragrunt init, validate, plan with no problems, however when I run terragrunt apply it stays at "helm_release.nginx: Still creating..." and at the end it gives me a timed out error.

do you have any idea about the reason of the error message?

I add images.

terragrunt int
![terragrunt init - kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182584-b11977c7-b591-418a-a360-dfa330061bd8.png)

terragrunt validate
![Terragrunt validate kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182617-d1cc419d-8f5c-4e13-bdfa-46ed046c056c.png)

terragrunt plan 
![terragrunt plan 1 kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182675-d49b1787-e445-4dc7-9bb6-b86b1b9866ce.png)

![terragrunt plan 2 kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182690-6266746b-056e-401d-b93c-6c860805aee6.png)

![terragrunt plan 3 kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182709-40a4963e-683d-4625-841d-f3402151f011.png)

![terragrunt plan 4 kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182739-bf1d33f7-5a3c-4fc9-b8c7-b89dcbe65d3e.png)

terragrunt apply
![terragrunt apply 1 kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182793-ca10597a-b725-4c2a-b3f1-d7aca764daa3.png)
![terragrunt apply 2 kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182814-ccef40b0-de39-4760-8700-b869fea38e1d.png)
![terragrunt apply 3 kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182824-96338abf-2671-4293-85b0-8dbfa0260280.png)
![terragrunt apply 4 kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182848-219c8a57-39a1-4552-a392-1504a5496d9b.png)

terragrunt apply error message
![terragrunt apply error kubernetes ingress](https://user-images.githubusercontent.com/102400906/167182896-410aee81-cbeb-4aae-98b3-edcd293c98ba.png)

**Comment by lpm0073** [comment]
Hello, confirming that the same thing happened to me yesterday. Re-running the apply operation a second time worked fine. It appears that I should extend the timeout period a few more minutes.

**Reply by Soporte-Crezee** [reply]
Hi Mr. Lawrence, I hope you are having a good week.

I tell you that I followed your suggestion, after the "timed out" message appeared i ran the "terragrunt apply" command a second time, but it didn't work, however i ran the apply command several times and got the same message error.

Could it be that the problem is in the time limit (5 minutes)?

In that case, How or in which file i change the time limit?

If the above is no the case, What could be happening?

*Update:

I put a time of 30 minutes, but in minute 21 it gave me the following error:

╷
│ Error: Kubernetes cluster unreachable: the server has asked for the client to provide credentials
│
│ with helm_release.nginx,
│ on main.tf line 50, in resource "helm_release" "nginx":
│ 50: resource "helm_release" "nginx" {
│
╵

I add image

![image](https://user-images.githubusercontent.com/102400906/167937986-03130687-1dc0-4777-a377-e65a1e8aa6d7.png)
